### PR TITLE
Expensive Bases Tests

### LINKED ISSUE #347: unit tests for large values of l and m for basis functions
Current the unit tests for basis functions using  small values of l and m.  We should add tests for large values so that the overflow problems will be identified more obvious when switching dtype from float64 to float32, such as shown in the line,
https://github.com/ComputationalCryoEM/ASPIRE-Python/blob/develop/src/aspire/basis/basis_utils.py#L129
https://github.com/ComputationalCryoEM/ASPIRE-Python/blob/develop/src/aspire/basis/basis_utils.py#L138

### PR DESCRIPTION
**Pull request by j-c-c**
Add expensive tests for bases to resolve #347

### CODE REVIEW

File: tests/test_FBbasis3D.py [hunk 1]
```diff
@@ -686,3 +685,31 @@ def testFBBasis3DExpand(self, basis):
             ],
             atol=utest_tolerance(basis.dtype),
         )
+
+
+# NOTE: This test is failing for L=64. `coeff_0` has a few NANs which propogate into `vol_1`...
```
Comment (garrettwrong) [inline tests/test_FBbasis3D.py]: Let's add the GH issue to the comment.

**Review by garrettwrong — DISMISSED** [review]
Only one small comment addition, but otherwise can move forward.

The FB 3D NaN is concerning, as is the poor tolerance when it works; wonder if related, but that's a different issue.  Thanks for adding the tests.

**Comment by garrettwrong** [comment]
Might want to squash the commits down while you're at it. Thanks.

File: tests/test_FBbasis3D.py (hunk 1 above)
Comment (j-c-c) [reply tests/test_FBbasis3D.py]: Done

**Review by garrettwrong — DISMISSED** [review]

File: tests/test_FBbasis2D.py [hunk 2]
```diff
@@ -134,3 +135,32 @@ def testComplexCoversionErrorsToReal(self, basis):
...
+        L=L,
+        dtype=dtype,
+        amplitudes=1,
+        offsets=0,
+        seed=seed,
+    )
+    im = sim.images[0]
+
+    # Round trip
+    coeff = basis.evaluate_t(im)
```
Comment (janden) [inline tests/test_FBbasis2D.py]: Shouldn't we be testing `expand`?

File: tests/test_FBbasis2D.py [hunk 3]
```diff
@@ -134,3 +135,32 @@ def testComplexCoversionErrorsToReal(self, basis):
...
+    FB_im = basis.evaluate(coeff)
```
Comment (janden) [inline tests/test_FBbasis2D.py]: `im_fb`

File: tests/test_FBbasis3D.py [hunk 4]
```diff
@@ -686,3 +685,31 @@ def testFBBasis3DExpand(self, basis):
...
+def testHighResFBbasis3D(L, dtype):
+    seed = 42
+    basis = FBBasis3D(L, dtype=dtype)
+    vol_0 = AsymmetricVolume(L=L, C=1, K=64, dtype=dtype, seed=seed).generate()
+
+    # First round trip
+    coeff_0 = basis.evaluate_t(vol_0)
+    vol_1 = basis.evaluate(coeff_0)
+
+    # Second round trip
```
Comment (janden) [inline tests/test_FBbasis3D.py]: Why two?

File: tests/test_FFBbasis2D.py [hunk 5]
```diff
@@ -213,3 +213,32 @@ def testShift(self, basis):
...
+        dtype=dtype,
+        amplitudes=1,
+        offsets=0,
+        seed=seed,
+    )
+    im = sim.images[0]
+
+    # Round trip
+    coeff = basis.evaluate_t(im)
+    FB_im = basis.evaluate(coeff)
```
Comment (janden) [inline tests/test_FFBbasis2D.py]: `im_ffb`

**Review by janden — CHANGES_REQUESTED** [review]

File: tests/test_FBbasis2D.py (hunk 2 above)
Comment (j-c-c) [reply tests/test_FBbasis2D.py]: Ah yes, thanks. That tightens up the tolerance quite a bit. Done.

File: tests/test_FBbasis2D.py (hunk 3 above)
Comment (j-c-c) [reply tests/test_FBbasis2D.py]: Done.

File: tests/test_FBbasis3D.py (hunk 4 above)
Comment (j-c-c) [reply tests/test_FBbasis3D.py]: When I was running this test for `L=55` I was getting a poor test tolerance. Taking a second round trip significantly improved that. I left in the test as it was when it was passing for L=55. If it's preferable I can remove the second round trip and adjust the tolerance in `np.allclose()`

File: tests/test_FFBbasis2D.py (hunk 5 above)
Comment (j-c-c) [reply tests/test_FFBbasis2D.py]: Done.

File: tests/test_FBbasis3D.py (hunk 4 above)
Comment (janden) [reply tests/test_FBbasis3D.py]: Ok but it's not clear to me why that should be the case. Have you tried `expand` here?

Comment (j-c-c) [reply tests/test_FBbasis3D.py]: Results with `expand` are similar. To be clear, I was comparing the result of the second round trip to the result of the first round trip (not to the original volume).

I think the best thing to do here would be to use a single round trip with `expand` and adjust the tolerance. The single round trip had comparable tolerances between `expand` and `evaluate_t`, with `expand` being slightly better.

Comment (j-c-c) [reply tests/test_FBbasis3D.py]: Made the changes as discussed in the meeting. Single round trip, using expand, with a skip for now.

**Review by janden — APPROVED** [review]
